mount_action: populate just one device in cache

nwf opened the pull request:
We're given the device basename as a parameter and then search through the cache to find it.  Just load the one device, saving the probes for the ones we don't care about.

FIXES https://github.com/openwrt/fstools/issues/2

dangowrt commented on block.c:
```diff
@@ -1253,7 +1253,13 @@ static int mount_action(char *action, char *device, int type)
 		return -1;
 
-	cache_load(1);
+	{
```
Instead of using `{ ... }` please just declare additional variables on top of the function OR create a new
`static void cache_load_dev(const char *device)` function.

nwf replied:
Like that?  (This spin is neither compile nor run tested.)

dangowrt commented on block.c:
```diff
@@ -595,6 +595,15 @@ static void cache_load(int mtd)
 	_cache_load("/dev/dm-*");
 }
 
+static int cache_load_dev_by_base_name(char *name)
```
make it `const char *name` and we are good to go ;)

nwf commented:
I believe this is ready to go, @dangowrt.  Yes?

nwf commented:
Ping?

nwf committed:
mount_action: populate just one device in cache

We're given the device basename as a parameter and then search through
the cache to find it.  Just load the one device, saving the probes for
the ones we don't care about.

FIXES https://github.com/openwrt/fstools/issues/2

Signed-off-by: Nathaniel Wesley Filardo <nwfilardo@gmail.com>
